packages/grid/_modules_/grid/components/default-footer.tsx

```diff
@@ -24,12 +25,26 @@ export const DefaultFooter = React.forwardRef<HTMLDivElement, DefaultFooterProps
       return null;
     }
 
+    const isPaginationAvailable = !!paginationComponent;
+    const showRowCount = !options.hideFooterRowCount && !isPaginationAvailable && (
+      <RowCount rowCount={totalRowCount} />
+    );
+    const showSelectedRowCount = !options.hideFooterSelectedRowCount && (
+      <SelectedRowCount selectedRowCount={selectedRowCount} />
+    );
+    const justifyItemsRight =
+      !selectedRowCount && !options.hideFooterSelectedRowCount && isPaginationAvailable;
+
     return (
-      <GridFooter ref={ref}>
-        {!options.hideFooterRowCount && <RowCount rowCount={totalRowCount} />}
-        {!options.hideFooterSelectedRowCount && (
-          <SelectedRowCount selectedRowCount={selectedRowCount} />
-        )}
+      <GridFooter
+        ref={ref}
+        className={classnames({
+          'MuiDataGrid-footer-pagination-visible': isPaginationAvailable,
```

Comment:
I think that it shouldatch the content, not the CSS logic we want to apply, it helps for customizations.

```suggestion
          'MuiDataGrid-footer-paginationAvailable': isPaginationAvailable,
```